agent/connectionuptime/service_test.go

```diff
@@ -267,3 +267,68 @@ func compareFloatWithTolerance(a, b float32) bool {
 
 	return diff < tolerance
 }
+
+func TestCalculationConnectionUpTimeWhenCleanupMethodIsNotCalled(t *testing.T) {
+	now := time.Now()
+	tests := []struct {
+		name             string
+		setOfConnections map[time.Time]bool
+		expectedUpTime   float32
+		windowPeriod     time.Duration
+		toTime           time.Time
+	}{
+		{
+			name: "should return 50% uptime when window period is 10s, and 100% uptime when window period is 5s",
+			setOfConnections: map[time.Time]bool{
+				now.Add(-8 * time.Second): false,
+				now.Add(-7 * time.Second): false,
+				now.Add(-6 * time.Second): false,
+				now.Add(-5 * time.Second): false,
+				now.Add(-4 * time.Second): true,
+				now.Add(-3 * time.Second): true,
+				now.Add(-2 * time.Second): true,
+				now.Add(-1 * time.Second): true,
+			},
+			expectedUpTime: 50,
+			windowPeriod:   10 * time.Second,
+			toTime:         now,
+		},
+		{
+			name: "should return 100% uptime when window period is 5s, and 100% uptime when window period is 10s",
```

Comment:
Same as above - we're not testing the 2nd part of the test name.

Reply:
fixed as well